File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic VB Programming/Inheritance/SalesPerson.vb
Public Class SalesPerson
    Inherits Employee

    'Class constructors
    Public Sub New(ByVal id As String, ByVal name As String, ByVal age As Integer, ByVal salary As Double, ByVal target As Integer)
        MyBase.New(id, name, age, salary)
        Me.target = target
    End Sub

    'Show polymorphism
    Public Overrides Sub awardBonus(ByVal bonus As Double)
        salary = salary + (bonus * 2)
    End Sub

    'Override ToString
    Public Overrides Function ToString() As String
        ToString = MyBase.ToString + " with a target of " + CStr(target) + vbNewLine
    End Function

    'Fields of Class
    Private target As Double
End Class

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic CS  Programming/Inheritance/SalesPerson.cs
using System;

namespace InheritanceDemo
{
	public class SalesPerson : Employee {	
		private double target;

		public SalesPerson(string id, string name, int age, double salary, double target) : base(id,name,age,salary) {	
			this.target = target;
		}

		//method must be override to enable polymorphism
		public override void awardBonus(double bonus) {
			salary += (bonus * 2);
		}

		public override string ToString() {
			return base.ToString() + " with target of " + target;
		}
	}
}
